### PR TITLE
[LLM Runtime] Enable GPTQ models

### PR DESCRIPTION
**Pull request by zhenwei-intel**
## Type of Change

feature

## Description

detail description 
JIRA ticket: [NLPTOOLKIU-931](https://jira.devtools.intel.com/browse/NLPTOOLKIU-931)

Support GPTQ model in llm runtime.

- [x] parse GPTQ model from inc
- [x] parse GPTQ model from huggingface
- [x] convert bloom GPTQ to q4_0
- [x] convert hugging face llama to q4_0
- convert to jblas will be done in next stage

## Expected Behavior & Potential Risk

```python
from transformers import AutoTokenizer, TextStreamer
from intel_extension_for_transformers.transformers import AutoModelForCausalLM, WeightOnlyQuantConfig

model_name = "/mnt/disk1/data2/zhenweil/models/gptq/Llama-2-7B-Chat-GPTQ"  # or local path to model
woq_config = WeightOnlyQuantConfig(from_gptq=True, use_cache=True)
prompt = "Once upon a time, a little girl"

tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)
inputs = tokenizer(prompt, return_tensors="pt").input_ids
streamer = TextStreamer(tokenizer)
model = AutoModelForCausalLM.from_pretrained(model_name, quantization_config=woq_config, trust_remote_code=True)

while True:
    prompt = input("> ").strip()
    if prompt == "quit":
        break
    b_prompt = "[INST]{}[/INST]".format(prompt)  # prompt template for llama2
    inputs = tokenizer(b_prompt, return_tensors="pt").input_ids
    outputs = model.generate(inputs, streamer=streamer, interactive=True, ignore_prompt=True, do_sample=True)

```

image

## How has this PR been tested?

N/A

## Dependency Change?

N/A

### CODE REVIEW

**Comment by kevinintel** [comment]
you can enable jblas in another PR

File: intel_extension_for_transformers/transformers/modeling/modeling_auto.py [hunk 1]
```diff
@@ -154,6 +154,7 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
                     use_ggml=quantization_config.use_ggml,
                     not_quant=quantization_config.not_quant,
                     use_cache=quantization_config.use_cache,
+                    from_gptq=quantization_config.from_gptq,
```
Comment (airMeng) [inline intel_extension_for_transformers/transformers/modeling/modeling_auto.py:157]: @xin3he do we usually have ```from_gptq``` field in the config?

File: intel_extension_for_transformers/llm/runtime/graph/application/main_pybind.cpp [hunk 2]
```diff
@@ -87,6 +91,32 @@ class Model {
...
+    int8_t* w_ptr = src_w.mutable_data();
+    float* scales_ptr = src_scales.mutable_data();
+    int8_t* dst_ptr = dst.mutable_data();
+
+    quant_params_internal q_params;
+    q_params.bits = quant_bits::q4;
+    q_params.scale_dtype = quant_sdtype::fp32;
+    q_params.compute_dtype = quant_comp::int8;
+    q_params.group_size = 128;
+    return Model::jblas_qpack(w_ptr, scales_ptr, nullptr, dst_ptr, q_params, 1, src_w.shape(0), src_w.shape(1));
```
Comment (airMeng) [inline intel_extension_for_transformers/llm/runtime/graph/application/main_pybind.cpp:108]: if leave to next pr, this should be removed or renamed?

File: intel_extension_for_transformers/transformers/modeling/modeling_auto.py (hunk 1 above)
Comment (hshen14) [reply intel_extension_for_transformers/transformers/modeling/modeling_auto.py:157]: Shall we unify the parameter? e.g., not_quant -> use_quant, and update the logic accordingly; from_gptq -> use_gptq?

File: intel_extension_for_transformers/llm/runtime/graph/application/main_pybind.cpp (hunk 2 above)
Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/application/main_pybind.cpp:108]: I will add jblas soon

**Review by a32543254 — APPROVED** [review]
LGTM

**Review by zhentaoyu — APPROVED** [review]

**Review by airMeng — APPROVED** [review]

**Pull request merged**